quantities/serialization/QuantityFactory.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Concurrent;
+using Quantities.Dimensions;
+using Quantities.Measures;
+
+namespace Quantities.Serialization;
+
+file static class Inject
```

Comment:
What is a `file static class`? I couldn't find any references for this.

Reply:
It's akin to `public`, `internal`, `private` etc. It means this class is public only within the file it is declared in. (The `static` just means it's a static class)

See also the docs [here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/file).